### PR TITLE
Adding config for local development

### PR DESCRIPTION
**Pull request by saf6260**
#### Summary
This PR addresses an ease-of-use as it pertains to running and building upon the base mattermost bot by providing a docker-compose environment 

The setup of this bot is documented very well in the README, but is time consuming, and may prohibit further developers from adding to the project. This PR attempts to outline/address this issue with the docker-compose file, as the number of commands needed to run the bot is greatly reduced

Further, this PR addresses issues with the mattermost password field, as new requirements require a special character and uppercase character for new user accounts.

Please feel free to close this PR if it is found to be unhelpful. However, I know for myself, it greatly assisted in getting my development environment up and running, and may help others in the future as well :)

#### Ticket Link
This PR is not related to a ticket link

### CODE REVIEW

**Comment by jfrerich** [comment]
@hanzei, is this something you could look at?

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+version: "3.9"
+services:
+  test_mm:
+    image: mattermost/mattermost-preview:5.26.0
+    container_name: test_mm
+    ports: 
+    - "8065:8065"
+
```
Comment (hanzei) [inline docker-compose.yaml:8]: Nit pick: Please add a line break

File: add_users.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+USER_NAME="bill"
+docker exec test_mm mattermost team create --name botsample --display_name "Sample Bot playground" --email "admin@example.com"
+docker exec test_mm mattermost user create --email="bot@example.com" --password="Password1!" --username="samplebot"
+docker exec test_mm mattermost user create --email="${USER_EMAIL}" --password="${USER_PASS}" --username="${USER_NAME}"
+docker exec test_mm mattermost roles system_admin bill
+docker exec test_mm mattermost team add botsample samplebot bill
+docker exec test_mm mattermost user verify samplebot
+
+echo "Default user credentials"
+echo "Email: ${USER_EMAIL} - Password: ${USER_PASS}"
```
Comment (hanzei) [inline add_users.sh:12]: Nit pick: Please add a line break

**Review by hanzei — APPROVED** [review]
LGTM :+1:

File: docker-compose.yaml (hunk 1 above)
Comment (saf6260) [reply docker-compose.yaml]: This has been added; good call!

File: add_users.sh (hunk 2 above)
Comment (saf6260) [reply add_users.sh]: This has also been added; good call!

**Review by catalintomai — APPROVED** [review]

**Pull request merged**